src/lib/payments.js

```diff
@@ -357,17 +353,29 @@ Payments.prototype.updateTransfer = function * (updatedTransfer, relatedResource
     return
   }
 
-  // Relevant source transfer, but not ready to settle yet.
-  if (this.settlementQueue.hasPaymentForTransfer(updatedTransfer.id)) {
-    return
-  }
-
   // TODO: should we delete the subscription?
   throw new UnrelatedNotificationError('Notification does not match a ' +
     'payment we have a record of or the corresponding source ' +
     'transfers may already have been executed')
 }
 
+Payments.prototype.updateSourceTransfer = function * (updatedTransfer, traderCredit) {
+  const destinationTransfer = traderCredit.memo
+  if (!destinationTransfer) return
+
+  // The source transfer isn't prepared yet, so ignore it.
+  if (updatedTransfer.state !== 'prepared' && updatedTransfer.state !== 'executed') {
```

Comment:
Can use a explaining variable here instead of a comment

```js
const isTransferReady = updatedTransfer.state === 'prepared' || updatedTransfer.state === 'executed
if (!isTransferReady) {
..
}
```